test/testcmdlineparser.cpp

```diff
@@ -368,14 +369,26 @@ class TestCmdlineParser : public TestFixture {
         ASSERT_EQUALS("", GET_REDIRECT_OUTPUT);
     }
 
-    void showversion() {
+    void version() {
         REDIRECT;
         const char * const argv[] = {"cppcheck", "--version"};
         ASSERT(parser->parseFromArgs(2, argv));
-        ASSERT_EQUALS(true, parser->getShowVersion());
-        ASSERT_EQUALS("", logger->str()); // version is not actually shown
+        ASSERT_EQUALS("Cppcheck 2.13 dev\n", logger->str());
     }
 
+    void versionWithCfg() {
+        REDIRECT;
+        ScopedFile file("cppcheck.cfg",
+                "{\n"
+                "\"productName\": \"The Product\""
+                "}\n");
+        const char * const argv[] = {"cppcheck", "--version"};
+        ASSERT(parser->parseFromArgs(2, argv));
+        ASSERT_EQUALS("The Product\n", logger->str()); // TODO: include version?
```

Comment:
The product name cannot contain the version number since it is being use as simply the product name is other places. I think we should include the version here.

As different builds might also make use of a different version we should have a way to specify this easily. I do not think the `cppcheck.cfg` would be the way to specify that.